logic/strategy/card_utils_test.go

```diff
@@ -0,0 +1,102 @@
+package strategy
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+)
+
+func isIn(c model.Card, h []model.Card) bool {
+	for _, thisCard := range h {
+		if c == thisCard {
+			return true
+		}
+	}
+	return false
+}
+
+func validateHand(origHand, thisHand []model.Card) bool {
+	for _, c := range thisHand {
+		if !isIn(c, origHand) {
+			return false
+		}
+	}
+	for i := 0; i < len(thisHand)-2; i++ {
+		// check that no duplicates exist in this hand
+		for _, c2 := range thisHand[i+1:] {
+			if thisHand[i] == c2 {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func factorial(n int) int {
```

Comment:
🌵 I know this is only used for tests, but could we use a memory cache of vars so that we don't have to recalc, say, 6! everytime we call `factorial(6)`, but just calc it once?